README.md

```diff
@@ -2,22 +2,406 @@
+
+`ddtrace` wraps Ruby code to take measurements of individual operations within your Ruby application. These measurements, called *spans*, are joined together to form a *trace*, which is sent to the Datadog agent.
+
+Here's an example of a manually traced web request:
+
+```ruby
+require 'ddtrace'
+
+get '/posts' do
+  Datadog.tracer.trace('web.request', service: 'my-blog', resource: 'GET /posts') do |span|
+    # Trace the query to the database
+    Datadog.tracer.trace('sql.query', service: 'blog-db') do
+      @posts = Posts.order(created_at: :desc).limit(10)
+    end
+
+    # Trace the template rendering
+    Datadog.tracer.trace('template.render') do
+      erb :index
+    end
+  end
+end
+```
+
+Which is then presented within Datadog as:
+
+**TODO: Add screenshot of flamegraph here.**
+
+Out of the box, `ddtrace` includes integrations for many popular Ruby gems and frameworks, to make it quick & easy to add instrumentation to your Ruby application.
 
 ## Getting started
 
-For a basic product overview, check out our [setup documentation][setup docs].
+Setting up tracing for your Ruby application takes only two steps:
+
+1. Setup the Datadog agent
+2. Configure your Ruby application
+
+### Setup the Datadog Agent
+
+Before adding the tracing library to your application, you will need to install the Datadog Agent, which receives trace data and forwards it to Datadog.
+
+Check out our instructions for how to [install and configure the Datadog Agent](https://docs.datadoghq.com/tracing/setup). You can also see additional documentation for [tracing Docker applications](https://docs.datadoghq.com/tracing/setup/docker).
+
+### Quickstart for Rails applications
+
+1. Add the `ddtrace` gem to your Gemfile:
+
+    ```ruby
+    source 'https://rubygems.org'
+    gem 'ddtrace'
+    ```
+
+2. Install the gem with `bundle install`
+3. Create a `config/initializers/datadog.rb` file containing:
+
+    ```ruby
+    Datadog.configure do |c|
+      # This will activate auto-instrumentation for Rails
+      c.use :rails
+    end
+    ```
+
+    Within this configuration block, you can also activate additional integrations here; (see [Integration instrumentation](#integration-instrumentation) for more information.)
+
+### Quickstart for Ruby applications
+
+1. Install the gem with `gem install ddtrace`
+2. Add a configuration block to your Ruby application:
+
+    ```ruby
+    require 'ddtrace'
+    Datadog.configure do |c|
+      # Configure the tracer here.
+      # Activate integrations, change tracer settings, etc...
+      # By default without additional configuration, nothing will be traced.
+    end
+    ```
+
+3. Add or activate instrumentation by doing either of the following:
+    - Activate integration instrumentation (see [Integration instrumentation](#integration-instrumentation))
+    - Add manual instrumentation around your code (see [Manual instrumentation](#manual-instrumentation))
+
+### Quickstart for OpenTracing
+
```

Comment:
I will suggest moving this content to this: https://docs.datadoghq.com/tracing/advanced/opentracing/?tab=ruby
and adding a link here.

Reply:
I'm in the process of reorganizing this; I'll share this later.